### PR TITLE
Feature/vmo 6851/vmo 7038/add toolbar

### PR DESCRIPTION
**Pull request by safydy-r**
Note for devs:
- We've amended the tech arch (see summary [here](https://miro.com/app/board/o9J_lXulmu4=/?moveToWidget=3458764534216067969&cot=14)), so we're creating a very minimal version of resource-viewer in flow-builder
- advantages: using vue-router "no reload time" feature, having one single toolbar, easy to maintain

[overview]
Resource-viewer menu not accessible by default
Screen Shot 2022-09-22 at 5 23 17 PM

Added a story to represent i
Screen Shot 2022-09-22 at 5 24 10 PM
t when visible though

### CODE REVIEW

File: src/components/interaction-designer/ResourceViewer.vue [hunk 1]
```diff
@@ -0,0 +1,67 @@
+<template>
+  <div class="resource-viewer panel panel-default">
+    <div class="panel-body resource-viewer-contents">
+      <h5>Blocks list:</h5>
```
Comment (safydy-r) [inline src/components/interaction-designer/ResourceViewer.vue:4]: no need to localize these texts, we will change this template in next subtasks

File: src/views/FetchFlow.vue [hunk 2]
```diff
@@ -71,7 +71,7 @@ class FetchFlow extends mixins(Routes, Lang) {
         await this.$router.replace(nextUrl)
       } else {
         this.message = 'flow-builder.flow-found'
-        this.flowLink = this.route('flows.editFlow', {flowId: this.activeFlow.uuid, component: 'designer', mode: 'edit'})
+        this.flowLink = this.route('flows.editFlow', {flowId: this.activeFlow.uuid, component: 'builder', mode: 'edit'})
```
Comment (safydy-r) [inline src/views/FetchFlow.vue:74]: renamed for more clarity. We have 02 components possible now:
- builder
- resource-viewer

File: src/views/InteractionDesigner.vue [hunk 3]
```diff
@@ -35,8 +35,14 @@
...
     <main class="interaction-designer-main">
       <builder-canvas
+        v-if="mainComponent === 'builder'"
         :width-adjustment="builderWidthAdjustment"
-        @click.native="handleCanvasSelected" />,
+        @click.native="handleCanvasSelected" />
+      <resource-viewer
+        v-else-if="mainComponent === 'resource-viewer'"/>
+      <div v-else>
+        <div class="alert alert-danger">ERROR: component '{{ mainComponent }}' is not supported</div>
```
Comment (safydy-r) [inline src/views/InteractionDesigner.vue:44]: for dev purpose in case we send a non-supported component

File: src/store/trees/trees.js [hunk 4]
```diff
@@ -78,7 +78,7 @@ export default {
     isFeatureTreeDuplicateEnabled: ({ui}) => lodash.find(ui.enabledFeatures, (feature) => feature === 'treeDuplicate'),
     isFeatureTreeViewVersionsEnabled: ({ui}) => lodash.find(ui.enabledFeatures, (feature) => feature === 'treeViewVersions'),
     isFeatureTreeDuplicateOfEnabled: ({ui}) => lodash.find(ui.enabledFeatures, (feature) => feature === 'treeDuplicateOf'),
-    isResourceEditorEnabled: ({ui}) => lodash.find(ui.enabledFeatures, (feature) => feature === 'resourceEditor'),
+    isResourceViewerEnabled: ({ui}) => lodash.find(ui.enabledFeatures, (feature) => feature === 'resourceViewer'),
```
Comment (safydy-r) [inline src/store/trees/trees.js:81]: renamed the feature to `resourceViewer` to reduce confusion, because we have component named `ResourceEditor.vue`

**Review by object-required — APPROVED** [review]

**Pull request merged**